lfric_macros/validate_rose_meta.py

```diff
@@ -0,0 +1,236 @@
+    failed = False
+
+    start_dir = os.path.join(source_path, "rose-stem", "app")
+    apps = os.listdir(start_dir)
+    for app in apps:
+        if app in INVALID_APPS:
+            continue
+        app_dir = os.path.join(start_dir, app)
+        conf_file = os.path.join(app_dir, "rose-app.conf")
+        with open(conf_file, "r") as f:
+            for line in f:
+                if line.startswith("meta="):
+                    break
+            else:
+                continue
+        command = f"rose macro --validate {meta_paths} -C {app_dir} --no-warn version"
+        result = run_command(command)
+        # Check that errors are only from rose-app-suite-controlled.conf
+        errors = []
+        if result.returncode:
+            errors = parse_suite_controlled(result.stderr)
+            errors = "\n".join(e for e in errors)
+
+        if errors:
+            print(f"[FAIL] - {app} failed to validate")
+            print(f"Failure validating app {app}:\n{errors}", file=sys.stderr)
+            failed = True
+        else:
+            print(f"[PASS] - {app} validated")
+
+    return failed
+
+
+def parse_args():
+    """
+    Read command line args
+    """
+
+    parser = argparse.ArgumentParser("Pre-process and apply LFRic Apps upgrade macros.")
+    parser.add_argument(
+        "-a",
+        "--apps",
+        default=None,
+        help="The path to the LFRic Apps source. At least one of this or Core are "
+        "required. If both are provided then it will be assumed that Apps is the "
+        "repository to check.",
+    )
+    parser.add_argument(
+        "-c",
+        "--core",
+        default=None,
+        help="The path to the LFRic Core source. At least one of this or Apps are "
+        "required. If both are provided then it will be assumed that Apps is the "
+        "repository to check.",
+    )
+
+    args = parser.parse_args()
+
+    if args.apps:
+        args.apps = os.path.expanduser(args.apps)
+    if args.core:
+        args.core = os.path.expanduser(args.core)
+
+    return args
+
+
+def main():
+    """
+    main function for this script
+    """
+    args = parse_args()
+
+    meta_paths = ""
+    rose_meta_path = ""
+    if args.apps:
+        source_path = args.apps
+        meta_paths += f"-M {os.path.join(args.apps, "rose-meta")}"
+        rose_meta_path += args.apps
+    elif args.core:
+        source_path = args.core
+        meta_paths += f"-M {os.path.join(args.core, "rose-meta")} "
+        if rose_meta_path:
+            rose_meta_path += f":{args.core}"
+        else:
+            rose_meta_path = args.core
+    else:
+        raise RuntimeError(
+            "At least one of the Apps or Core sources must be provided as a command "
+            "line argument"
+        )
+
+    failures = False
+
+    if check_rose_metadata(rose_meta_path, source_path):
+        failures = True
+
+    if check_rose_stem_apps(meta_paths, source_path):
+        failures = True
+
+    if failures:
```

Comment:
```suggestion

    if check_rose_metadata(rose_meta_path, source_path) or
       check_rose_stem_apps(meta_paths, source_path):
```
This should give the same result.